leader/src/jerigon.rs

```diff
@@ -27,21 +27,24 @@ pub(crate) async fn jerigon_main(
     )
     .await?;
 
-    let block_proofs = prover_input
+    // We keep only last block proof from the interval. It contains
+    // all the necessary information to verify the whole interval.
+    let last_block_proof = prover_input
         .prove(&runtime, previous_proof, save_inputs_on_error)
-        .await?;
+        .await?
+        .into_iter()
+        .last()
+        .ok_or_else(|| anyhow!("No block proof generated"))?;
     runtime.close().await?;
 
-    for block_proof in block_proofs {
-        let block_proof_str = serde_json::to_vec(&block_proof)?;
-        write_proof(
-            block_proof_str,
-            proof_output_dir_opt.clone().map(|mut path| {
-                path.push(format!("b{}.zkproof", block_proof.b_height));
-                path
-            }),
-        )?;
-    }
+    let last_block_proof_serialized = serde_json::to_vec(&last_block_proof)?;
+    write_proof(
+        last_block_proof_serialized,
+        proof_output_dir_opt.clone().map(|mut path| {
+            path.push(format!("b{}.zkproof", last_block_proof.b_height));
+            path
+        }),
+    )?;
```

Comment:
Partly related to the design question above, if we go in this direction, then we _may_ want to flush the previous proof passed as arg as well (up to the user). Could be configurable ideally.

Reply:
Yes, I had the same idea. Will update accordingly